test/config.test.js

```diff
@@ -122,4 +122,20 @@ describe('config', () => {
       expect(configs).toMatchSnapshot();
     });
   });
+
+  test('load: mode with no config', () => {
+    const argv = { mode: 'development' };
+    const config = undefined;
+    return load(argv, { config }).then((configs) => {
+      expect(configs).toMatchSnapshot();
+    });
+  });
+
+  test('load: mode overrides config', () => {
+    const argv = { mode: 'production' };
+    const config = './test/fixtures/webpack-4-defaults/webpack.config.js';
```

Comment:
this is a good test, but we also need a test with the multi-config to make sure the mode was applied correctly to an array of configs